spring-cloud-sleuth-zipkin/src/test/java/org/springframework/cloud/sleuth/zipkin/ZipkinSpanListenerTests.java

```diff
@@ -135,6 +139,45 @@ public void rpcAnnotations() {
 		assertEquals(2, this.test.spans.size());
 	}
 
+	@Test
+	public void appendsLocalComponentTagIfNoZipkinLogIsPresent() {
+		this.parent.logEvent("hystrix/retry");
+		this.parent.stop();
+
+		zipkin.Span result = this.listener.convert(this.parent);
+
+		assertThat(result.binaryAnnotations)
+				.extracting(input -> input.key)
+				.contains(Constants.LOCAL_COMPONENT);
+	}
+
+	@Test
+	public void appendsServerAddressTagIfClientLogIsPresent() {
+		this.parent.logEvent(Constants.CLIENT_SEND);
+		this.parent.stop();
+
+		zipkin.Span result = this.listener.convert(this.parent);
+
+		assertThat(result.binaryAnnotations)
+				.filteredOn("key", Constants.SERVER_ADDR)
+				.extracting(input -> input.value)
+				.containsOnly("unknown".getBytes(Charset.forName("UTF-8")));
+	}
+
+	@Test
+	public void shouldReuseServerAddressTag() {
+		this.parent.logEvent(Constants.CLIENT_SEND);
+		this.parent.tag(Constants.SERVER_ADDR, "fooservice");
```

Comment:
I understand why you are re-using this, but it doesn't make as much sense here as it looks like. the value of SERVER_ADDR is a struct that includes the IP address. I'd use the tag "peer.service" from OpenTracing in sleuth and in the zipkin converter, make that either a CLIENT_ADDR or SERVER_ADDR instead. https://github.com/opentracing/opentracing-go/blob/master/ext/tags.go#L21

In the zipkin v2, I'd expect to process "peer.service" directly. I've already started playing with this here: https://github.com/openzipkin/zipkin-java/blob/master/zipkin/src/main/java/zipkin/internal/DependencyLinkSpan.java